ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -868,9 +866,55 @@ export const RuntimePanel = fp.flow(
 
     return runtimeRequest;
   };
+
+  // 50 GB is the minimum GCP limit for disk size, 4000 GB is our arbitrary limit for not making a
+  // disk that is way too big and expensive ($.22 an hour)
+  const validators = {
+    selectedDiskSize: {
+      numericality: {
+        greaterThanOrEqualTo: 50,
+        lessThanOrEqualTo: 4000,
+        message: '^Disk size must be between 50 and 4000 GB'
+      }
+    }
+  };
+  // We don't clear dataproc config when we change compute type so we can't combine this with the
+  // above or else we can end up with phantom validation fails
+  const dataprocValidators = {
+    masterDiskSize: {
+      numericality: {
+        greaterThanOrEqualTo: 50,
+        lessThanOrEqualTo: 4000,
+        message: 'must be between 50 and 4000 GB'
+      }
+    },
+    workerDiskSize: {
+      numericality: {
+        greaterThanOrEqualTo: 50,
+        lessThanOrEqualTo: 4000,
+        message: 'must be between 50 and 4000 GB'
+      }
+    }
+  };
+  const errors = validate({selectedDiskSize}, validators);
+  const {masterDiskSize = null, workerDiskSize = null} = selectedDataprocConfig || {};
+  const dataprocErrors = selectedCompute === ComputeType.Dataproc
```

Comment:
is it necessary for `errors` and `dataprocErrors` to be separate? it looks like they're often used together. I'm wondering if we can just push the `dataprocErrors` into the original `errors` variable.

Reply:
Yes, I have a comment about it above but I can move it below - we only update `dataprocConfig` when we call `updateRuntime`. This is so that if you switch away from dataproc compute type and back, your config hasn't been forgotten. If the validators are combined, the following steps will cause the create/next button to be disabled when it shouldn't be:

1. Select Dataproc config type
2. Choose 49 for worker disk size
3. Select Standard config type

because `dataprocConfig.workerDiskSize` is still 49 until the actual update is performed.

Reply:
got it. sounds good